docs/madr/2-sqlite.md

```diff
@@ -0,0 +1,89 @@
+# Using aos-sqlite Module
+
+- Status: proposed
+- Deciders: [Dylan], [Ariel], [Phil]
+- Date: 2024-07-22
+- Authors: [Atticus]
+
+## Context and Problem Statement
+
+The `aos-sqlite` module is a WebAssembly (Wasm) binary in the Arweave (ao) ecosystem that extends the functionality of the standard `aos` Wasm binary module. The `aos` module includes an `Eval` handler that consumes Lua strings and uses the `load` method of Lua to execute them. The `aos-sqlite` module is compiled with `lua-sqlite3` from C to Wasm, providing additional SQLite functionality. Given that directly using tables has demonstrated scalability issues, the SQLite implementation offers a potential solution. However, due to its novelty in the ecosystem, there are concerns about its stability in larger products. Therefore, using it here for the ANT Registry provides a suitable environment seperate from other products to test its stability.
+
+## Decision Drivers
+
+The main drivers for this decision are:
+
+- **Scalability**
+  - The current method of using tables has scalability issues that SQLite can potentially address.
+- **Performance**
+  - Improved performance in handling larger datasets with SQLite.
+- **Risk Management**
+  - Mitigating risks in larger products by testing the module in a controlled environment first.
+- **Ecosystem Integration**
+  - Exploring new modules and their potential benefits within the ao ecosystem.
+
+## Considered Options
+
+### Option 1: Continue Using Existing Tables
+
+- **Pros:**
+  - Known and stable.
+  - No need for additional testing or integration.
+- **Cons:**
+  - Demonstrated scalability issues.
+  - Limited performance with larger datasets.
+  - Lack of strictness on data types
+
+### Option 2: Use aos-sqlite Module
+
+- **Pros:**
+  - Addresses scalability issues with direct table usage.
+  - Potential for improved performance with larger datasets.
+  - Maintained by core team of AO
+- **Cons:**
+  - New and relatively untested in the ecosystem.
+  - Potential stability issues in larger products.
+
+## Decision Outcome
+
+It was decided to use the `aos-sqlite` module to address scalability issues in a controlled environment. This approach allows us to test its effectiveness and performance before considering wider adoption in larger products.
```

Comment:
What is the "controlled environment"?

Reply:
I believe it refers to "in this less-than-mission-critical process" (i.e. the ANT Registry). However, I'm not comfortable with the idea that we'd be ok with this experiment failing there. We really want to give ourselves the best chance of this working long term.

Reply:
Yes, the intended meaning is that the ANT Registry is seperated from more mission-critical products like the ANT modules and Network process which our other systems rely upon. The ANT registry is apart from those and if IT goes down, other applications like the ar.io gateway and arns resolver would still be functional.

I am also not comfortable with it failing here, just wanted to identify that this is unknown territory for us from a development standpoint. Given the other testing done in https://github.com/ar-io/ar-io-ant-registry-process/pull/2 im more confident in it now that I tested it more deeply and worked with it, but it still seems worth noting I think.

Reply:
Made a brief explanation of that.